src/Spectre.libHeatmapDataScaling/HistogramEqualization.h

```diff
@@ -0,0 +1,49 @@
+/*
+* HistogramEqualization.h
+* Class with histogram equalization algorithm implementation for contrast enhancement.
+*
+Copyright 2018 Daniel Babiak
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+#include <map>
+#include "Spectre.libHeatmapDataScaling/HeatmapDataScalingAlgorithm.h"
+
+namespace spectre::visualization
+{
+/// <summary>
+/// Class containing histogram equalization algorithm implementation.
+/// </summary>
+class HistogramEqualization : public HeatmapDataScalingAlgorithm
```

Comment:
The whole class relies on an implicit assumption, that values of intensity do not exceed the value 255. If this is really a thing, then I would consider putting it in some global constexpr in the header file to avoid writing magic numbers.